relay-config/src/config.rs

```diff
@@ -968,6 +972,26 @@ pub struct Processing {
     /// Maximum rate limit to report to clients.
     #[serde(default = "default_max_rate_limit")]
     pub max_rate_limit: Option<u32>,
+    /// The number of logical partitions that can receive flushed buckets.
+    ///
+    /// If set, buckets are partitioned by (bucket key % partitions), and routed
+    /// by setting the header `X-Sentry-Relay-Shard`.
+    #[serde(default)]
+    pub metrics_partitions: Option<u64>,
+    /// The approximate maximum number of bytes submitted in one metrics batch.
+    ///
+    /// This controls how big flushed batches of buckets get, depending on the number of buckets,
+    /// the cumulative length of their keys, and the number of raw values. Since final serialization
+    /// adds some additional overhead, this number is approxmate and some safety margin should be
+    /// left to hard limits.
+    #[serde(default = "default_metrics_max_batch_size")]
+    pub metrics_max_batch_size: ByteSize,
+    /// The approximate maximum number of bytes submitted in one metrics batch for processing
+    /// relays.
+    ///
+    /// Overrides [`Self::metrics_max_batch_size`] when specified on processing relays.
+    #[serde(default)]
+    pub metrics_max_batch_size_processing: Option<ByteSize>,
```

Comment:
Maybe should set the default to 100KiB what it currently is (configured in ops configs)